### PR TITLE
[FIX] payment, payment_custom: uninstall crash of payment modules

### PR DESCRIPTION
**Pull request by hote-odoo**
Before this commit a constraint between the `ir_ui_view` and `payment_provider` view made impossible for providers to be uninstalled.
After this commit the constraint is correctly treated.

A similar issue is found on a constraint on the `payment_custom` module. To fix this and future possible problems we added the possibility to modify the way providers are uninstalled following each provider needs.

Task - 3002532

See also:
- https://github.com/odoo/documentation/pull/2799

### CODE REVIEW

File: addons/payment/models/payment_provider.py [hunk 1]
```diff
@@ -564,6 +564,11 @@ def _remove_provider(self, provider_code):
         providers.write({
             'code': 'none',
             'state': 'disabled',
+            'redirect_form_view_id': False,
+            'inline_form_view_id': False,
+            'token_inline_form_view_id': False,
+            'express_checkout_form_view_id': False,
```
Comment (Feyensv) [inline addons/payment/models/payment_provider.py:570]: :+1:

File: addons/payment_custom/models/payment_provider.py [hunk 2]
```diff
@@ -38,6 +38,14 @@ def _compute_view_configuration_fields(self):
             'show_cancel_msg': False,
         })
 
+    def _get_provider_removal_values(self):
+        res = super()._get_provider_removal_values()
+
+        if self.code == 'custom':
+            res['custom_mode'] = False
```
Comment (Feyensv) [inline addons/payment_custom/models/payment_provider.py:45]:
```suggestion
        res['custom_mode'] = False
```

`self` is empty here.  IMHO, you can always set custom_mode to False, it won't be a problem.

**Comment by Feyensv** [comment]
Red ci/style

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**